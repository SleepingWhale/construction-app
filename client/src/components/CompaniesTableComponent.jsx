import React from 'react';

import PropTypes from 'prop-types';


export const CompaniesTableComponent = ({ companies }) => {
  return (
    <table className="table">
      <thead>
      <tr>
        <th scope="col"/>
        <th scope="col">Company name</th>
        <th scope="col">City</th>
        <th scope="col">Specialty</th>
      </tr>
      </thead>
      <tbody>
      {
        companies.map(({
         id, picture, companyName, city, specialty,
        }) => (
          <tr key={id}>
            <td>
              <img src={picture} alt={companyName} className="rounded"/>
            </td>
            <td>{companyName}</td>
            <td>{city}</td>
            <td>
              {
                specialty.map((s, i) => (<div key={i}>{s}</div>))
              }
            </td>
          </tr>
        ))
      }
      </tbody>
    </table>
  );
};

CompaniesTableComponent.propTypes = {
  companies: PropTypes.arrayOf(PropTypes.shape(
    {
      id: PropTypes.string,
      picture: PropTypes.string,
      companyName: PropTypes.string,
      city: PropTypes.string,
      specialty: PropTypes.arrayOf(PropTypes.string)
    }
  ))
};

CompaniesTableComponent.defaultProps = {
  companies: [],
};
