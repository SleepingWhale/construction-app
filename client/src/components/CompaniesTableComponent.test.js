import React from 'react';
import { shallow } from 'enzyme';
import { CompaniesTableComponent } from './CompaniesTableComponent';


const companies = [
  {
    id: '4deccc6a-8f3a-4cf2-bb87-5cf5b1874a11',
    picture: 'http://placekitten.com/100/100',
    companyName: 'Cubicide',
    city: 'Concho',
    specialty: [
      'Excavation',
      'Electrical'
    ]
  },
  {
    id: 'd1ab1c52-3a02-4582-8277-7aafae6a7cd9',
    picture: 'http://placekitten.com/100/100',
    companyName: 'Diginetic',
    city: 'Iola',
    specialty: [
      'Electrical'
    ]
  },
  {
    id: 'edae29c1-6afc-42d8-9923-9eb3715c443f',
    picture: 'http://placekitten.com/100/100',
    companyName: 'Fibrodyne',
    city: 'Hachita',
    specialty: [
      'Excavation',
      'Electrical',
      'Plumbing'
    ]
  }
];


describe('CompaniesTableComponent', () => {
  it('should render correctly', () => {
    const component = shallow(<CompaniesTableComponent companies={companies}/>);
    expect(component).toMatchSnapshot();
  });
});
