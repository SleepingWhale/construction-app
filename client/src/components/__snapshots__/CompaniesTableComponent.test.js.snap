// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompaniesTableComponent should render correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Found: undefined
      </th>
      <th
        scope="col"
      >
        Company name
      </th>
      <th
        scope="col"
      >
        City
      </th>
      <th
        scope="col"
      >
        Specialty
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="4deccc6a-8f3a-4cf2-bb87-5cf5b1874a11"
    >
      <td>
        <img
          alt="Cubicide"
          className="rounded"
          src="http://placekitten.com/100/100"
        />
      </td>
      <td>
        Cubicide
      </td>
      <td>
        Concho
      </td>
      <td>
        <div
          key="0"
        >
          Excavation
        </div>
        <div
          key="1"
        >
          Electrical
        </div>
      </td>
    </tr>
    <tr
      key="d1ab1c52-3a02-4582-8277-7aafae6a7cd9"
    >
      <td>
        <img
          alt="Diginetic"
          className="rounded"
          src="http://placekitten.com/100/100"
        />
      </td>
      <td>
        Diginetic
      </td>
      <td>
        Iola
      </td>
      <td>
        <div
          key="0"
        >
          Electrical
        </div>
      </td>
    </tr>
    <tr
      key="edae29c1-6afc-42d8-9923-9eb3715c443f"
    >
      <td>
        <img
          alt="Fibrodyne"
          className="rounded"
          src="http://placekitten.com/100/100"
        />
      </td>
      <td>
        Fibrodyne
      </td>
      <td>
        Hachita
      </td>
      <td>
        <div
          key="0"
        >
          Excavation
        </div>
        <div
          key="1"
        >
          Electrical
        </div>
        <div
          key="2"
        >
          Plumbing
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
